### PR TITLE
v8.1.2: bump version

### PR DESCRIPTION
**Pull request by Oreoxmt**
### What is changed, added or deleted? (Required)

v8.1.2: bump version

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [ ] master (the latest development version)
- [ ] v8.5 (TiDB 8.5 versions)
- [ ] v8.4 (TiDB 8.4 versions)
- [ ] v8.3 (TiDB 8.3 versions)
- [ ] v8.2 (TiDB 8.2 versions)
- [x] v8.1 (TiDB 8.1 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/19320
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: functions-and-operators/json-functions.md [hunk 1]
```diff
@@ -128,9 +128,9 @@ The subsequent content takes the following JSON document as an example to demons
 | `$.database.features[0]`              | The first database feature.             | `"distributed"`               |
 | `$.database.features[2]`              | The third database feature.             | `"relational"`                |
 | `$.database.versions[0].type`         | The type of the first database version. | `"lts"`                       |
-| `$.database.versions[*].release_date` | The release date for all versions.      | `["2024-08-27","2024-03-29"]` |
+| `$.database.versions[*].release_date` | The release date for all versions.      | `["2024-12-20","2024-03-29"]` |
 | `$.*.features`                        | Two array's of features                 | `[["distributed", "scalable", "relational", "cloud native"], ["MySQL compatible", "Shard merging"]]`                              |
-| `$**.version`                         | All versions, with path wildcard        | `["v8.1.1","v8.0.0"]`         |
+| `$**.version`                         | All versions, with path wildcard        | `["v8.1.2","v8.0.0"]`         |
```
Comment (hfxsd) [inline functions-and-operators/json-functions.md:133]: 这里说 All versions，需要加上 v8.1.1 吗？
```suggestion
| `$**.version`                         | All versions, with path wildcard        | `["v8.1.2","v8.1.1","v8.0.0"]`         |
```

File: functions-and-operators/json-functions.md [hunk 2]
```diff
@@ -128,9 +128,9 @@ The subsequent content takes the following JSON document as an example to demons
 | `$.database.features[0]`              | The first database feature.             | `"distributed"`               |
 | `$.database.features[2]`              | The third database feature.             | `"relational"`                |
 | `$.database.versions[0].type`         | The type of the first database version. | `"lts"`                       |
-| `$.database.versions[*].release_date` | The release date for all versions.      | `["2024-08-27","2024-03-29"]` |
+| `$.database.versions[*].release_date` | The release date for all versions.      | `["2024-12-20","2024-03-29"]` |
```
Comment (hfxsd) [inline functions-and-operators/json-functions.md:131]:
```suggestion
| `$.database.versions[*].release_date` | The release date for all versions.      | `["2024-12-20","2024-08-27","2024-03-29"]` |
```

File: tiup/tiup-component-cluster-patch.md [hunk 3]
```diff
@@ -29,7 +29,7 @@ Before running the `tiup cluster patch` command, you need to pack the binary pac
 1. Determine the following variables:
 
     - `${component}`: the name of the component to be replaced (such as `tidb`, `tikv`, or `pd`).
-    - `${version}`: the version of the component (such as `v8.1.1` or `v7.5.3`).
+    - `${version}`: the version of the component (such as `v8.1.2` or `v7.5.3`).
```
Comment (hfxsd) [inline tiup/tiup-component-cluster-patch.md:32]:
```suggestion
    - `${version}`: the version of the component (such as `v8.1.2` or `v7.5.4`).
```

File: functions-and-operators/json-functions.md (hunk 1 above)
Comment (Oreoxmt) [reply functions-and-operators/json-functions.md:133]: 上面示例就是 v8.1.2 和 v8.0.0
image

**Review by hfxsd — APPROVED** [review]

File: upgrade-tidb-using-tiup.md [hunk 4]
```diff
@@ -74,10 +74,11 @@ This section introduces the preparation works needed before upgrading your TiDB
 
 Review compatibility changes in TiDB release notes. If any changes affect your upgrade, take actions accordingly.
 
-The following provides compatibility changes you need to know when you upgrade from v8.0.0 to the current version (v8.1.1). If you are upgrading from v7.6.0 or earlier versions to the current version, you might also need to check the compatibility changes introduced in intermediate versions in the corresponding [release notes](/releases/release-notes.md).
+The following provides compatibility changes you need to know when you upgrade from v8.0.0 to the current version (v8.1.2). If you are upgrading from v7.6.0 or earlier versions to the current version, you might also need to check the compatibility changes introduced in intermediate versions in the corresponding [release notes](/releases/release-notes.md).
 
 - TiDB v8.1.0 [compatibility changes](/releases/release-8.1.0.md#compatibility-changes)
 - TiDB v8.1.1 [compatibility changes](/releases/release-8.1.1.md#compatibility-changes)
+- TiDB v8.1.2 [compatibility changes](/releases/release-8.1.2.md#compatibility-changes)
```
Comment (Oreoxmt) [inline upgrade-tidb-using-tiup.md:81]:
```suggestion
- TiDB v8.1.2 [release notes](/releases/release-8.1.2.md)
```

**Pull request merged**